### ISSUE TITLE
the RXDONE irq may be missing when kernel close global IRQ

### ISSUE THREAD

**Issue by wx-ta** [fifteenhex/sx127x#1]
As we know,The kernel will close the global irq for some thing needed,If the sx127x received a packet and generate a irq when global irq is off.  so the driver's code that clear sx127x RXDONE irq cannot be called, does this case leads RXDONE signal(sx127x DIO0 Pin) push high level forever? does it leads the driver cannot receive packet forever?

**Comment by fifteenhex** [comment]
I haven't touched the driver in a while. I plan to finish it off in the next few weeks.
I'll test this situation and get back to you.

**Comment by wx-ta** [comment]
I have test the driver on linux 3.14(i.mx6ul)：
1、Cut off the wire which connects the sx1278 DIO0 pin to the IMX6 IRQ pin；
2、Send a packet  to the device，then i got high level on DIO0 pin，and never fall down；
3、Reconnect the IRQ wire, then the DIO pin fall down ，because the i.mx6 irq pin received a raising edge interrupt，and the driver code clears the IRQ flag register ；

SO，I add 2 ioctl functions in the driver code：

**case SX127X_IOCTL_CMD_GETIRQ:
	ret = sx127x_getirq(data);
	break;	
case SX127X_IOCTL_CMD_CLEARIRQ:
	ret = sx127x_clearirq(data,arg);
	break;**


**static int sx127x_getirq(struct sx127x *data){
    u8 irqflags;
	sx127x_reg_read(data->spidevice, SX127X_REG_LORA_IRQFLAGS, &irqflags);
    return irqflags;
}


static int sx127x_clearirq(struct sx127x *data, unsigned int irq_bits){
    u8 regv;
	regv = irq_bits & 0xff;
	sx127x_reg_write(data->spidevice, SX127X_REG_LORA_IRQFLAGS, regv);
    return 0;
}**

I try to check IRQ register periodic manually in user space，if pending status be found twice then clear the register with writing 0xff ；
I think this is the easiest way to solve this problem but not the best way！Do anyone have a better idea？

**Comment by fifteenhex** [comment]
I don't think you should need to check the IRQ register in userland. I'll add a periodic check in the driver itself.

**Comment by wx-ta** [comment]
That is a good solution !
